lib/shopify_api/resources/fulfillment_order_fulfillment.rb

```diff
@@ -0,0 +1,4 @@
+module ShopifyAPI
+  class FulfillmentOrderFulfillment < Base
```

Comment:
Wondering if this is needed? I'm thinking that the fulfillments returned via `GET fulfillment_orders/:fulfillment_order_id/fulfillments` shouldn't be treated any differently than the existing fulfillment object.

Reply:
It wasn't clear at the time whether it was ok to mix/match existing Fulfillment actions with those newly added, seemed like it might get confusing to remember which are legacy type requests vs new fulfillment order ones.

Reply:
Should we restrict apps from calling legacy requests on fulfillments (https://help.shopify.com/en/api/reference/shipping-and-fulfillment/fulfillment#show-2019-10) associated to fulfillment orders? I don't think we should since non opted-in FS should be able to query for fulfillments from a fulfillment order and still run legacy requests.

Reply:
There are a few conflicting overlaps that have to get resolved if we collapse them into a single `Fulfullment` resource. Queries will either route to under orders or fulfillment_orders. There's also the `cancel` action that's defined for both legacy and FO fulfullment.

Reply:
We could collapse them together, and have to override a lot of the default `ActiveResource` behaviour to include both types of actions (legacy and new). It may not be necessary to do all of that extra work though, talked with Malaz and it should be fine to return a fulfillment resource that doesn't support legacy actions. Non opted-in FS shouldn't really even be querying fulfillments from fulfillment orders (even though we don't restrict them from doing so).

Reply:
If we know the long term desired end state then we can choose a path towards it. What is the expected adoption of FOs and support for legacy operations. Based on that, what would be clearest for consumers of the API and other supporting documentation, to have a single Fulfillment resource or the legacy and V2?

Reply:
The long term desired end state is that all apps adopt FOs and will not be using the legacy operations. Consumers who use operations that return `FulfillmentV2` should be opted-in and thus should be using the new `V2` operations and are blocked from using legacy operations. It wouldn't be worth trying to merge the two together, just so in the future we undo all of that work. Documentation is not very clear on the distinction between the two though (both are called Fulfillment resources, one is just under the unstable version and the other latest version).

Reply:
There's no technical problem with returning `Fulfillment` in the shopify_api from a v2 operation. Only confusion would be then trying to perform a legacy operation on it. It really comes down to how the two types of Fulfillment resources get documented. If we distinguish the operations as legacy / FO we can use the same `Fulfillment` resource in shopify_api. If we document two different types of Fulfillment resources then we can call the new one FulfillmentV2.

Reply:
Currently, the documentation exists like so:
* https://help.shopify.com/en/api/reference/shipping-and-fulfillment/fulfillment#complete-2019-10
* https://help.shopify.com/en/api/reference/shipping-and-fulfillment/unstable/fulfillmentunstable

Reply:
Thanks this is very clear. We have two separate descriptions of distinct Fulfillment resources. We should go with `FulfillmentV2`

Reply:
@karmakaze I'm working on documentation right now and realized that the `shipping-and-fulfillment/unstable/fulfillmentunstable` doc will be moved to under the `shipping-and-fulfillment/fulfillment` doc (under the RC version). The new `Fulfillment` resource under the 2020-01 version will have both the legacy operations and new operations that we are adding. With this, should we combine FulfillmentV2 and Fulfillment?

Reply:
Yes, it makes sense to combine them to follow the documentation structure.